### DISCUSSION TITLE
should I use 'this' keyword in getters ?

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [vuejs/pinia#703] [Help and Questions] [answered]
I just came across pinia a few days ago, and I am still learning the basics. In the official [docs](https://pinia.esm.dev/core-concepts/getters.html) they don't use `this` in the getters, but then I saw this [post](https://engineering.wedevs.com/2021/04/23/pinia-a-lightweight-vuex-alternative/) and they use `this` in getters and actions, is there any benefits in doing so ?

**Comment by posva** [comment, marked as answer]
When you use this in a getter you need to explicitly type the return type of it.
